operator/v1/types_network.go

```diff
@@ -267,6 +267,11 @@ type OVNKubernetesConfig struct {
 	// +kubebuilder:validation:Minimum=0
 	// +optional
 	MTU *uint32 `json:"mtu,omitempty"`
+	// geneve port is the UDP port to be used by geneve encapulation.
+	// Default is 6081
+	// +kubebuilder:validation:Minimum=0
```

Comment:
to be sure: is zero really a valid port?

Reply:
@sttts No. 0 is not valid. We use 0 here and in most integer fields to make sure there are no negative numbers. The actual checks are in the code. @danwinship

Reply:
why not 1? users can set zero and things don't work

Reply:
@stts When people set 0 or 70000 etc. things don't work. On AWS if they pick other than 6081 or a port in the range 9000-9999 that is not currently used it will break as well.  Look at the rest of the api, 0 is not useful for most fields, just makes sure its a uint.

Reply:
IMO it's a bad user experience and oneliners away

Reply:
@stts I understand and agree with what you are saying. There is only so much that can be done at compile time and there are a large number of constraints in selecting a usable port. Can we continue with 0 since it is the common practice in this file?

Reply:
It's a new field. It is the only chance ever to properly validate it without breaking API compatibility and improve the user experience. Just because we did a mistake a number of times before is a bad argument for doing it again.

Reply:
Changed min to 1
@sttts @danwinship @squeed @dcbw ( LGTM label has been removed.) PTAL

Reply:
The `Minimum=0` is not there to perform validation, it's there to work around a bug in controller-tools; if we don't put it there, then controller-tools will generate a CRD that lets people create `network.operator.openshift.io` resources that can't be unmarshaled into a `networkoperatorv1.Network` and so can't even be seen by CNO.

The plan was that once controller-tools got fixed to handle unsigned types correctly, we would remove all of the `Minimum=0` annotations since they would be no-ops at that point.

If you insist, we can put `Minimum=1, Maximum=65535` here, but it seems confusing to me to have half of the validation here and half of it in CNO. (And we can't have it all here because kubebuilder/OpenAPI isn't expressive enough to fully validate the object.)